spec/views/two_factor_authentication/otp_verification/show.html.slim_spec.rb

```diff
@@ -258,35 +287,31 @@
 
     context 'when users phone number is unconfirmed' do
       it 'has a link to choose a new phone number' do
-        data = presenter_data.merge(
-          unconfirmed_phone: true,
-          reenter_phone_number_path: 'some/path'
-        )
+        data = presenter_data.merge(unconfirmed_phone: true)
 
-        @presenter = TwoFactorAuthCode::PhoneDeliveryPresenter.new(data)
+        @presenter = TwoFactorAuthCode::PhoneDeliveryPresenter.new(
+          data: data,
+          view: view
+        )
 
         render
 
-        expect(rendered).to have_link(
-          t('forms.two_factor.try_again'), href: @presenter.reenter_phone_number_path
-        )
+        expect(rendered).to have_link(t('forms.two_factor.try_again'), href: verify_phone_path)
```

Comment:
two reasons:

1) before` reenter_phone_number_path` was a public method, but I made it private since it is only used within the class ( so can no longer call `@presenter.reenter_phone_number_path`)

2) before, we were using `attributes_for(:generic_otp_presenter)` in this spec file, which for me was causing a mystery guest problem. I changed it out for a hash above, where I am using `verify_phone_path` instead of a random string (which is what it was before). Can switch back if you like, just thought it was nice to have more realistic data.